### PR TITLE
spirv-fuzz: Do not allow adding stores to read-only pointers

### PR DESCRIPTION
**Pull request by afd**
Adds checks for PushConstant and UniformConstant storage classes (in
addition to the existing check for Input storage class).

### CODE REVIEW

File: source/fuzz/transformation_store.cpp [hunk 1]
```diff
@@ -125,5 +126,18 @@ protobufs::Transformation TransformationStore::ToMessage() const {
...
+  switch (storage_class) {
+    case SpvStorageClassInput:
+    case SpvStorageClassPushConstant:
+    case SpvStorageClassUniformConstant:
+      // Read-only: cannot store to it.
+      return true;
+    default:
+      return false;
+  }
+}
```
Comment (paulthomson) [inline source/fuzz/transformation_store.cpp:140]: There seems to already be `IsReadOnlyVariableShaders`. Could that be used? It looks more complex though.

Comment (afd) [reply source/fuzz/transformation_store.cpp:140]: It appears that it only applies to OpVariable instructions.  I need to know whether a pointer is suitable for a store, and that pointer could have come from an access chain.

Looks like the code should be adaptable, though.  I have opened an issue (#3324) which I will reference from this code, and will refactor later if it turns out to be reusable.

Comment (afd) [reply source/fuzz/transformation_store.cpp:140]: I have made a PR - #3325 - that should allow us to re-use this functionality.

Comment (afd) [reply source/fuzz/transformation_store.cpp]: Adapted this PR to re-use that functionality.

**Pull request merged**